src/application-development/bootloader.md

```diff
@@ -1,4 +1,41 @@
-# Building a Custom ESP-IDF Bootloader
+# Application Startup Flow
+
+In order to boot an application, Espressif devices use 2 bootloaders:
+- First Stage Bootloader (ROM Bootloader): Sets up architecture-specific registers, checks the [boot mode][boot-mode] and reset reason, and loads the second stage bootloader.
+- Second Stage Bootloader: Loads the [partition table][partition-table] and your application.
```

Comment:
This might be a left over from before, because I can see you've split up the esp-idf bits below nicely. To improve this bit I have some suggestions.

The role of the second stage bootloader is:

* To probe what RAM/PSRAM/FLASH is connected, and set those up
* Various misc things like setting up watchdogs, and some basic clocks such that we can boot into the main application